src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -1007,45 +1027,45 @@ void VM_Version::get_processor_features() {
   }
 
   if (!UseAPX) {
-    _features &= ~CPU_APX_F;
+    _features.clear_feature(CPU_APX_F);
   }
 
   if (UseAVX < 2) {
-    _features &= ~CPU_AVX2;
-    _features &= ~CPU_AVX_IFMA;
+    _features.clear_feature(CPU_AVX2);
+    _features.clear_feature(CPU_AVX_IFMA);
   }
 
   if (UseAVX < 1) {
-    _features &= ~CPU_AVX;
-    _features &= ~CPU_VZEROUPPER;
-    _features &= ~CPU_F16C;
-    _features &= ~CPU_SHA512;
+    _features.clear_feature(CPU_AVX);
+    _features.clear_feature(CPU_VZEROUPPER);
+    _features.clear_feature(CPU_F16C);
+    _features.clear_feature(CPU_SHA512);
   }
 
   if (logical_processors_per_package() == 1) {
     // HT processor could be installed on a system which doesn't support HT.
-    _features &= ~CPU_HT;
+    _features.clear_feature(CPU_HT);
   }
 
   if (is_intel()) { // Intel cpus specific settings
     if (is_knights_family()) {
-      _features &= ~CPU_VZEROUPPER;
-      _features &= ~CPU_AVX512BW;
-      _features &= ~CPU_AVX512VL;
-      _features &= ~CPU_AVX512DQ;
-      _features &= ~CPU_AVX512_VNNI;
-      _features &= ~CPU_AVX512_VAES;
-      _features &= ~CPU_AVX512_VPOPCNTDQ;
-      _features &= ~CPU_AVX512_VPCLMULQDQ;
-      _features &= ~CPU_AVX512_VBMI;
-      _features &= ~CPU_AVX512_VBMI2;
-      _features &= ~CPU_CLWB;
-      _features &= ~CPU_FLUSHOPT;
-      _features &= ~CPU_GFNI;
-      _features &= ~CPU_AVX512_BITALG;
-      _features &= ~CPU_AVX512_IFMA;
-      _features &= ~CPU_AVX_IFMA;
-      _features &= ~CPU_AVX512_FP16;
+      _features.clear_feature(CPU_VZEROUPPER);
```

Comment:
Should we be also clearing the CPU_AVX10_1 and CPU_AVX10_2 here?

Reply:
I agree; it may help validate KNL on Diamond Rapids :-)